src/main/java/eu/stamp_project/mutationtest/descartes/operators/ThisMutationOperator.java

```diff
@@ -7,40 +7,31 @@
 import org.pitest.reloc.asm.commons.Method;
 
 /**
- * Replaces the method body with a <code>return this</code> statement
+ * Replaces the method body with a <code>return null</code> statement
  */
 public class ThisMutationOperator extends MutationOperator{
 
     /**
      * Returns a value indicating whether the operator can transform the given method.
-     * In this case whether <code>this</code> could be assigned to the return type.
+     * In this case whether <code>null</code> could be assigned to the return type.
      *
      * @param method Method to be tested by the operator
-     * @return A boolean value indicating if this can be assigned to the return type
+     * @return A boolean value indicating if null can be assigned to the return type
      */
-
-    @Override
-    public boolean canMutate(Method method) {
-        return true;
-    }
-
     @Override
     public boolean canMutate(ClassName className, Method method) {
-        int target = method.getReturnType().getSort();
         String[] objArr = method.getReturnType().toString().split("/");
         String[] classArr = className.asJavaName().split("\\.");
         int check = objArr[objArr.length - 1].indexOf(classArr[classArr.length - 1]);
```

Comment:
Aren't you here comparing only the last part of the name?
What will happen if we have 
```
className -> a.b.AClass
method.getReturnType -> c.d.AClass
```
These two classes `a.b.AClass` and `c.d.AClass` will have the same simple name but they are not the same class as they belong to two different packages.

On the other hand, the comparison searches for the index. Is there any special reason for that? What would happen if we compare `AClass` and `AClassB`, the `indexOf` will return a non-negative value and the comparison would be wrong.